task2/task2.js

```diff
@@ -0,0 +1,41 @@
+//a
+//window.x = 1;
+const bind =
+  (fn, context) =>
+  (...args) =>
+    fn.apply(context, [...args]);
+const context = { x: 2 };
+const testThis = function (y) {
+  console.log(`x=${this.x}, y=${y}`);
+};
+testThis(100); // x=1, y=100
+
+const boundFunction = bind(testThis, context);
+console.log(boundFunction(100)); // x=2, y=100
+
+//b
+const Robot = function (name) {
+  this.name = name;
+};
+function add(op1, op2) {
+  this.name = this.name || "Human";
+  return this.name + " can count to " + (op1 + op2);
+}
+const voltron = new Robot("Voltron");
+
+("Human can count to 1"); //0 + 1
+console.log(add(0, 1));
+console.log(add.bind(this)(0, 1));
+console.log(add.apply(this, [0, 1]));
+console.log(add.call(this, 0, 1));
+
+("Voltron can count to 3"); //1 + 2
+let a = add.bind(voltron);
```

Comment:
What `a` variable mean here? What other engineers will think about it? Why `let` here?